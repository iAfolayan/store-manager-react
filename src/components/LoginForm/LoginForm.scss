label.text-sm-left {
  float: left;
}

.reset-password {
  color: #fff;
  display: inline;
  line-height: 35px;
}

.login{
  margin-left: 2rem
}

// Small devices (landscape phones, 576px and up)
@media (max-width: 576px) {
  .wrapper {
    width: 85%;
  }
  .app-name {
    font-size: 1rem;
  }
 }

// Medium devices (tablets, 768px and up)
@media (min-width: 768px) {
  .wrapper {
    width: 85%;
  }
 }

// Large devices (desktops, 992px and up)
@media (min-width: 992px) {
  .wrapper {
    width: 40%;
  }
 }

// Extra large devices (large desktops, 1200px and up)
@media (min-width: 1200px) {
  .wrapper {
    width: 30%;
  }
  .reset-password {
    color: #fff;
    display: block;
    margin-left: 0.75rem;
  }
  
  .reset-password>a {
    font-size: 12px;
    text-transform: uppercase;
    color: #fff
  }
 }