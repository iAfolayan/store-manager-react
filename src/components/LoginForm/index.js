import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { LogInValidator } from '../../helpers/validate';
import './LoginForm.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userLogin } from '../../actions/authActions/authActions';
import InputForm from '../InputForms';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        staffId: '',
        password: ''
      },
      isLoading: false,
      errors: {}
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const { user } = this.state;
    const errors = LogInValidator(user);
    if (errors) {
      this.setState({ errors });
    }
    const { login } = this.props;
    login(user);
  }

  handleChange(event) {
    const { user } = this.state;
    const { name, value } = event.target;
    user[name] = value;
    this.setState({
      user
    });
  }

  render() {
    const { user, errors, isLoading } = this.state;
    const { authIsLoading } = this.props.auth;
    if (this.props.auth.isAuthenticated) { return <Redirect to="/products" />}
    return (
      <form onSubmit={this.handleSubmit}>
        <InputForm
          error={errors.staffId}
          text="StaffId"
          label="Staff Id"
          type="text"
          name="staffId"
          value={user.staffId.toUpperCase()}
          placeholder="Enter staff Id"
          onChange={this.handleChange}
      />
        <InputForm
          error={errors.password}
          text="Password"
          label="Password"
          type="password"
          name="password"
          value={user.password}
          placeholder="Enter staff password"
          onChange={this.handleChange}
      />
        <div className="d-flex flex-row-reverse">
          <button onClick={this.handleChange} type="submit" className="btn btn-outline-primary login" disabled={authIsLoading ? isLoading: null}>{authIsLoading ? 'Loading...' : 'Login'}</button>
          <span className="d-flex align-items-center reset-password mt10">
Forget password? &nbsp;
            <Link to="/password"> Click here</Link>
          </span>
        </div>
      </form>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  login: user => dispatch(userLogin(user))
});

const mapStateToProps = state => ({
  auth: state.authReducer
});

export { LoginForm as Login };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginForm);

LoginForm.propTypes = {
  login: PropTypes.func,
  auth:PropTypes.object,
};

LoginForm.defaultProps = {
  login: null,
  auth: {}
}
