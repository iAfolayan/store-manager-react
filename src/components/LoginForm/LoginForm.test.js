import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import LoginForm, { LoginForm as LoginFormUnit} from '.';


describe('<LoginForm />', () => {
  test('should render login form', () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should simulate on submit event', () => {
    const wrapper = shallow(<LoginForm />);
    wrapper.instance().handleSubmit({
      preventDefault: jest.fn()
    });
    expect(wrapper).toBeDefined();
  });

  test('should simulate handleChange event', () => {
    const newValue = 'SM0002';
    const wrapper = mount(
      <Router>
        <LoginForm />
      </Router>);
    wrapper.find('input[type="text"]').simulate('change', {
      target: { name: newValue }
    });
    wrapper.update();
    expect(wrapper.find('input[type="text"]').prop('name')).toContain('staffId');
    expect(wrapper.find('input[type="password"]').prop('name')).toContain('password');
  });

  test('should redirect if user is authenticated', () => {
    props.auth.isAuthenticated = true;
    const wrapper = shallow(<LoginFormUnit {...props} />);
    expect(wrapper.is(Redirect)).toEqual(true);
  });

  test('should login the user after providing logon information', () => {
    const wrapper = mount(<LoginFormUnit {...props} />);

    const staffId = wrapper.find('input[name="staffId"]');
    const password = wrapper.find('input[name="password"]');
    const loginForm = wrapper.find('button[type="submit"]');

    staffId.simulate('change', { taget: { value: 'SM001' } });
    password.simulate('change', { taget: { value: 'blahbalh' } });
    loginForm.simulate('submit');

    expect(props.handleSubmit).toHaveBeenCalled();
  });
});
