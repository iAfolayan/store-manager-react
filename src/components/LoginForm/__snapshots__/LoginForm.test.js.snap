// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render login form 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
